tkg/client/common.go

```diff
@@ -369,7 +381,7 @@ func (c *TkgClient) ShouldDeployClusterClassBasedCluster(isManagementCluster boo
 		log.Warning(constants.YTTBasedClusterWarning)
 	}
 
-	if config.IsFeatureActivated(constants.FeatureFlagForceDeployClusterWithClusterClass) {
+	if c.IsFeatureActivated(constants.FeatureFlagForceDeployClusterWithClusterClass) {
```

Comment:
What 's this feature flag used for? Can you draft a doc about the combination of all related flags, variables?

Reply:
`FeatureFlagForceDeployClusterWithClusterClass` is used to deploy classy based cluster even if user has ytt customization to the provider templates. You can find most of feature gates definition [here](https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/constants/featureflags.go).